### PR TITLE
Add interfaces to UnitToken and BaseToken, etc.

### PR DESCRIPTION
**Pull request by timesaved12345**
Add IUnitToken, IBaseToken, IBurnable, IMintable.
Move all events and errors to interfaces.

### CODE REVIEW

**Comment by ivebeenherebefore** [comment]
Overall, this looks pretty good to me

File: contracts/MineToken.sol
Comment (StarDumpling) [inline contracts/MineToken.sol:1]: I think we're missing `IMineToken` interface.

File: contracts/auctions/UnitAuction.sol [hunk 1]
```diff
@@ -56,7 +56,7 @@ contract UnitAuction is IUnitAuction, Proxiable, Ownable {
     IBondingCurve public immutable bondingCurve;
     IERC20 public immutable collateralToken;
     uint256 private immutable collateralTokenDecimals;
-    UnitToken public immutable unitToken;
+    IUnitToken public immutable unitToken;
```
Comment (StarDumpling) [inline contracts/auctions/UnitAuction.sol:59]: We can do the same with `unitToken` and `mineToken` in the `BondingCurve` contract.

File: contracts/MineToken.sol
Comment (timesaved12345) [reply contracts/MineToken.sol:1]: Done.

File: contracts/auctions/UnitAuction.sol (hunk 1 above)
Comment (timesaved12345) [reply contracts/auctions/UnitAuction.sol:59]: Done.

**Pull request merged**